src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -368,13 +368,13 @@ internal unsafe SocketError DoOperationReceiveFromSingleBuffer(SafeCloseSocket h
         {
             fixed (byte* bufferPtr = &MemoryMarshal.GetReference(_buffer.Span))
             {
-                Debug.Assert(_singleBufferHandleState == SingleBufferHandleState.None);
-                _singleBufferHandleState = SingleBufferHandleState.InProcess;
-                var wsaBuffer = new WSABuffer { Length = _count, Pointer = (IntPtr)(bufferPtr + _offset) };
-
                 NativeOverlapped* overlapped = AllocateNativeOverlapped();
                 try
                 {
+                    Debug.Assert(_singleBufferHandleState == SingleBufferHandleState.None);
+                    _singleBufferHandleState = SingleBufferHandleState.InProcess;
+                    var wsaBuffer = new WSABuffer { Length = _count, Pointer = (IntPtr)(bufferPtr + _offset) };
```

Comment:
Is there a functional reason for moving these inside of the try?

Reply:
It needs to be after the AllocateNativeOverlapped call, which means it's either just before the try or just inside it. And as there's no downside to being just inside it, doing so helps highlight visually what's protected by the try, with the allocation just before it. But, technically, since there aren't thread aborts and that code isn't allocating, it would probably be ok if it were outside the try.